src/mmrelay/migrate.py

```
@@ -0,0 +1,2588 @@
+        required_bytes = MIN_FREE_SPACE_BYTES
+
+    # Use parent directory if path is a file or doesn't exist
+    check_path = path
+    if not check_path.exists() or check_path.is_file():
+        check_path = check_path.parent
+
+    try:
+        usage = shutil.disk_usage(str(check_path))
+        # Add 50% safety margin
+        required_with_margin = int(required_bytes * 1.5)
+        return (usage.free >= required_with_margin, usage.free)
```

Ruff (0.14.14)

[warning] Consider moving this statement to an `else` block

(TRY300)